float32 current1
float32 current2
float32 current3